src/txn/txn_ckpt.c

```diff
@@ -1975,6 +2093,9 @@ __wt_checkpoint_close(WT_SESSION_IMPL *session, bool final)
     if (ret == 0 && !F_ISSET(btree, WT_BTREE_SKIP_CKPT))
         ret = __checkpoint_tree(session, false, NULL);
 
+    /* Do not store the cached checkpoint list when checkpointing a single file alone. */
+    __wt_meta_saved_ckptlist_free(session);
+
```

Comment:
Why do we have two calls here in the same function?

Reply:
The `__checkpoint_tree()` call will end up saving a checkpoint list, which might be okay, to be honest. But I am being overly cautious with this change and discarding the saved checkpoints before and after individual checkpoint through `__wt_checkpoint_close`, and `__wt_checkpoint()`.

Another alternative could be to introduce a bool `btree->ckptlist_invalid`, and set it to true. The next use of the saved checkpoint list could then get rid of the list and go slow path. If that sounds a cleaner approach, I can give it a try, let me know.

Reply:
@sueloverso , what do you think? If we decide to introduce `btree->ckptlist_invalid`, I will rather create another ticket so that we can merge this change sooner.

Reply:
Happy with smaller follow-on tickets to get this merged.

Reply:
Created WT-7521 to work on this comment. Marking resolved for now.